neutron_plugin/router.py

```diff
@@ -148,14 +148,11 @@ def dict_merge(a, b):
             'API returned unexpected structure.: {0}'.format(router))
 
     router_id = router['router'].pop('id')
-    new_router = {
-        'router': {
-            'admin_state_up': args.get('admin_state_up', True),
-            'routes': args.get('routes', []),
-            'external_gateway_info': args.get(
-                'external_gateway_info', {}),
-        }
-    }
+
+    new_router = {'router': {}}
+    if len(args.keys()) > 0:
```

Comment:
Is this condition really necessary? The following `for` loop will simply not be executed if `len(args.keys())` is zero.

Reply:
Removed.